Assets/_BForBoss/_Core/Scripts/Environment/PlayerTriggerAreaEffect.cs

```diff
@@ -0,0 +1,127 @@
+using Perigon.Character;
+using Perigon.Entities;
+using Perigon.Utility;
+using Sirenix.OdinInspector;
+using UnityEngine;
+
+namespace BForBoss
+{
+    [RequireComponent(typeof(Collider))]
+    public class PlayerTriggerAreaEffect : MonoBehaviour
+    {
+        private enum PlayerEffect
+        {
+            Damage,
+            Heal
```

Comment:
Why do you have 2 separate ones. 
it'll be confusing to have 2 different enums. MIght as well put them all under PlayerEffect.

Reply:
Okay I see the issue here. 
So the biggest issue is that `PlayerEffect` and `PlayerModifier` is basically the same meaning to others, and can get confusing. 

if u want to keep it as two enums, you can, but you have to be explicit to these meanings. like
`enum HealthChange` `enum SpeedChange` 

Or second thing, is to create a struct with the mutators. 
```
[System.Serializable]
public struct Mutator {
     enum PlayerEffect { Health, Speed }
    float percentage
}
```

And then you pass in this mutator in via inspector as an array of mutator. This is more future proof, but a bit more work.

Reply:
Went ahead and made it a class instead so that I can add default values to certain fields!